src/replication.c

```diff
@@ -2931,6 +2936,10 @@ void syncWithMaster(connection *conn) {
 }
 
 int connectWithMaster(void) {
+    server.repl_master_sync_attempts++;
+    if (server.repl_master_connect_time == 0)
+        server.repl_master_connect_time = server.unixtime;
```

Comment:
since every time a replica getting disconnected or getting configured as a replica, the replica will trigger connect, so the `repl_master_connect_time` can indicate `disconnect` and cover all the cases.

Reply:
it can trigger that re-connection with some delay (next replicationCron).
i think that logically it should be in the disconnection.

Reply:
you mentioned in the meeting that moving this to the disconnection would cause complications, i don't see why, can you explain?

i agree that every disconnection will lead to a re-connection, but unless it really simplifies things, i argue that it's logically odd to put it here.